homeassistant/components/smart_meter_texas/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow for Smart Meter Texas integration."""
+import asyncio
+import logging
+
+from aiohttp import ClientError
+from smart_meter_texas.async_api import Auth, SMTError
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required("meter"): str,
+        vol.Required("esiid"): str,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    client_session = aiohttp_client.async_get_clientsession(hass)
+
+    try:
+        auth = Auth(client_session, data["username"], data["password"])
+        await auth.authenticate()
+    except (asyncio.TimeoutError, ClientError):
+        raise CannotConnect
+    except SMTError:
+        raise InvalidAuth
+
+    # Return info that you want to store in the config entry.
+    return {"title": "Electric Meter"}
```

Comment:
We should use something to identify the meter that is human readable in case they have multiple meters.

The meter api returns the `address` so that might make sense here

Reply:
Ideally we create sensors for all their meters so the title might make more sense to associate with the account / username

Reply:
I'm not exactly sure what the API would return if the user has multiple meters. I would have to see what JSON is returned in that case.


This is the format returned for one meter:
```
{
    "data": {
        "defaultMeterDetails": {
            "description": "123 Fake St",
            "address": "123 FAKE ST",
            "fullAddress": "123 FAKE ST, DALLAS, TX, 75226-0.00",
            "esiid": "12345678901234567",
            "meterNumber": "123456789",
            "meterMultiplier": 1
        },
        "latestDate": "07/21/2020",
        "reportPref": "INTERVAL",
        "count": 1,
        "meterReadDate": "07/21/2020 00:00:00",
        "lastMeterReading": "9017.875",
...
}
```

Reply:
I'll add another meter to my account and report back

Reply:
Here is the response from `Request URL: https://www.smartmetertexas.com/api/meter`


```json
{
	"data": [{
		"customer": "X",
		"email": "Y",
		"description": "REMOVED",
		"address": "REMOVED",
		"city": "HOUSTON",
		"state": "TX",
		"zip": "77018-1111",
		"esiid": "REMOVED",
		"meterNumber": "REMOVED",
		"meterMultiplier": 1,
		"fullAddress": "REMOVED",
		"errmsg": "Success",
		"recordStatus": "ESIID-RECORD-EXISTS",
		"statusIndicator": true,
		"dunsNumber": "NO_DUNS"
	}, {
		"customer": "X",
		"email": "Y",
		"description": "REMOVED",
		"address": "REMOVED",
		"city": "HOUSTON",
		"state": "TX",
		"zip": "77007-",
		"esiid": "REMOVED",
		"meterNumber": "REMOVED",
		"meterMultiplier": 1,
		"fullAddress": "REMOVED",
		"errmsg": "Success",
		"recordStatus": "ESIID-RECORD-EXISTS",
		"statusIndicator": true,
		"dunsNumber": "NO_DUNS"
	}]
}
```

Reply:
Oh I see. I hadn't looked at that endpoint. It looks to be a better way to get the individual meters than the dashboard.